iothub/device/src/Transport/Mqtt/MqttIotHubAdapter.cs

```diff
@@ -858,7 +863,7 @@ private async Task SendMessageAsync(IChannelHandlerContext context, Message mess
             }
 
             PublishPacket packet = await ComposePublishPacketAsync(context, message, qos, topicName).ConfigureAwait(true);
-            var publishCompletion = new TaskCompletionSource();
+            var publishCompletion = new DotNetty.Common.Concurrency.TaskCompletionSource();
```

Comment:
Should we use the TaskCompletionSource we have above or DotNetty one? If we need to use DotNetty one here, can you add a code comment for it?

Reply:
I'll add a comment for this, sure